### PR TITLE
Ci/rebuild base image

### PR DESCRIPTION
**Pull request by simonpintarelli**
trying to get cscs pipeline rebuild the base images

### CODE REVIEW

**Comment by toxa81** [comment]
I think we still need one build with MKL. It can be without elpa.

**Comment by simonpintarelli** [comment]
Here for example? 
```
RUN spack env create -d /sirius-env-cuda && \
    spack -e /sirius-env-cuda add  "sirius@develop %gcc build_type=RelWithDebInfo +tests +apps +cuda +magma ^mpich ^intel-oneapi-mkl threads=openmp" && \
    spack -e /sirius-env-cuda develop -p /sirius-src sirius@develop && \
    spack -e /sirius-env-cuda install --only=dependencies --fail-fast
```
I'll wait for the current run to finish, it seems it does not cache docker layers and restarts from scratch.

**Comment by toxa81** [comment]
> Here for example?
> 
> ```
> RUN spack env create -d /sirius-env-cuda && \
>     spack -e /sirius-env-cuda add  "sirius@develop %gcc build_type=RelWithDebInfo +tests +apps +cuda +magma ^mpich ^intel-oneapi-mkl threads=openmp" && \
>     spack -e /sirius-env-cuda develop -p /sirius-src sirius@develop && \
>     spack -e /sirius-env-cuda install --only=dependencies --fail-fast
> ```
> 
> I'll wait for the current run to finish, it seems it does not cache docker layers and restarts from scratch.

I pushed slighly different version with explicit scalapack.

**Comment by simonpintarelli** [comment]
some rocblas dependency failed with:
```
    280    checking whether program_invocation_name is defined... yes
     281    checking whether program_invocation_short_name is defined... yes
     282    checking for pidfd_open... no
     283    checking for __NR_pidfd_open... yes
     284    checking for ncursesw... no
     285    checking for ncurses... no
  >> 286    configure: error: ncurses support missing/incomplete (for partial b
            uild use --without-ncurses)
```
Seems ubuntu changed ncurses package upstream which was picked up by spack external find

**Comment by simonpintarelli** [comment]
cscs-ci run

File: README.md [hunk 1]
```diff
@@ -293,3 +293,4 @@ The development of the SIRIUS library would not be possible without support of t
 |![pasc](doc/images/logo_marvel.png) | NCCR MARVEL <br> Centre on Computational Design and Discovery of Novel Materials | https://nccr-marvel.ch/ |
 |![pasc](doc/images/logo_max.png)    | MAX (MAterials design at the eXascale) <br> European Centre of Excellence | http://www.max-centre.eu/   |
 |![pasc](doc/images/logo_prace.png)  | Partnership for Advanced Computing in Europe | https://prace-ri.eu/  |
+
```
Comment (simonpintarelli) [inline README.md:296]: Let's remove this whitespace again before merging

**Comment by simonpintarelli** [comment]
@toxa81, I'm confused too, the timings on the CI per test didn't  look unusually high. Is it possible jfrog is lagging when it tries to pull the image. The size is 12GB, afaik same as before.

**Comment by toxa81** [comment]
oh...come on! now there is no scalapack eigen-solver in scalapack build

**Comment by simonpintarelli** [comment]
Sorry! I was not careful with the specs.

**Comment by toxa81** [comment]
> Sorry! I was not careful with the specs.

No worries, Simon! 
I believe it was correct. I checked the output of "build cuda image" job which builds "/sirius-env-cuda-mkl-mpich". All was set properly.

File: ci/baseimage.cuda.Dockerfile
Comment (toxa81) [inline ci/baseimage.cuda.Dockerfile:1]: I think this should be rolled back. For running on daint we use
```
build cuda image:
  extends: .container-builder
  needs: ["build base cuda image"]
  stage: build
  variables:
    DOCKERFILE: ci/build.Dockerfile
    PERSIST_IMAGE_NAME: $CSCS_REGISTRY_PATH/sirius/sirius-ci:$CI_COMMIT_SHA
    ENVPATH: "/sirius-env-cuda-mkl-mpich"
    DOCKER_BUILD_ARGS: '["BASE_IMAGE=${BASE_IMAGE}", "ENVPATH=$ENVPATH"]'
    
.run_tests:
  extends: .container-runner-daint-gpu
  needs: ["build cuda image"]
  ```

**Comment by simonpintarelli** [comment]
> > Sorry! I was not careful with the specs.
> 
> No worries, Simon! I believe it was correct. I checked the output of "build cuda image" job which builds "/sirius-env-cuda-mkl-mpich". All was set properly.

Indeed, then I broke it with the last commit. I've removed it via force push, lets hope it will pickup the prebuilt container again.

File: ci/baseimage.cuda.Dockerfile
Comment (simonpintarelli) [reply ci/baseimage.cuda.Dockerfile:1]: Yes, exactly `/sirius-env-cuda` is not run, only `/sirius-env-cuda-mkl-mpich` is.

**Comment by simonpintarelli** [comment]
Seems the pitfall was that both
`build cuda image openblas` (aka `/sirius-env-cuda`) and `build cuda image` (aka `/sirius-env-cuda-mkl-mpich`) pushed the container to jfrog.
https://github.com/electronic-structure/SIRIUS/blob/ci/rebuild-base-image/ci/cscs-daint.yml#L30

Tests seems running fine now.

**Comment by toxa81** [comment]
I agree that JFrog might be lagging. The failures happen randomly

**Comment by toxa81** [comment]
Both hang in band parallel test for He.

**Comment by toxa81** [comment]
cscs-ci run default

**Comment by simonpintarelli** [comment]
cscs-ci run default

**Comment by toxa81** [comment]
@simonpintarelli fantastic work!

**Pull request merged**